[17.0][MIG] l10n_es_aeat: Migration to 17.0

ramiadavid opened the pull request:
Depends on:

- [x] depends on https://github.com/OCA/account-financial-reporting/pull/1087

Commit:
[ADD] l10n_es_aeat, l10n_es_aeat_347, l10n_es_aeat_349: Adds a generic module for aeat models, 347 module was portedto v6.0 and adds new module to print AEAT model 349.

Commit:
[IMP+FIX] l10n_es_aeat, l10n_es_aeat_mod347, l10n_es_aeat_mod349: Pequeñas mejoras generales (vistas, traducciones, código) y corrección pequeño bug por un olvido en la adaptación de la v5 a la v6

albertca committed:
[FIX] l10n_es_aeat: Fix _format_number function so number '22027.0' is formatted correctly. With this number 'dec_part' would become 100 (instead of 0) and fail in the final assertion. Instead of providing a workaround for this case, I simply replace the formatting with '%0*.*f' which seems easier specially because it already does the appropriate rounding.

Commit:
[REF] Cambio de licencia de GPL3 a AGPL en los módulos donde sólo aparecía Pexego en la licencia, quedan los compartidos.

Commit:
[FIX] l10n_es_aeat_mod347,l10n_es_aeat: Añadimos los últimos cambios en la 6.0 de estos módulos

Commit:
Commit inicial de una rama para la versión 7.0. Todos los módulos fueron marcados como no instalables mientras no se compruebe que funcionan o migren, para poder usarlos o probarlos se debería poner el atributo installable de los ficheros __openerp__.py de cada módulo a True

acysos committed:
[ADD] l10n_es_aeat: migración a OpenERP 7.0

Commit:
[FIX] l10n_es_aeat: fix exception call (except_orm instead of orm_exception)

acysos committed:
[FIX] l10n_es_aeat: bug atthach_id

Commit:
[FIX] msgstr in .pot files should be empty

Commit:
[IMP] l10n_es_aeat: Icono para el módulo.
- Algo de PEP8 y refactorización de código

[MIG] l10n_es_aeat_mod303: Migración a v7 del módulo con las siguientes mejoras:
- Convenciones de la comunidad.
- PEP8 en la mayoría de código.
- Icono para el módulo.
- Adaptación de vistas a las posibilidades de la v7.
- Restringidas las cuentas bancarias a las de la compañía.

Commit:
[IMP] l10n_es_aeat: Mejorado mecanismo de herencia para reaprovechar más código.
- Añadido wizard para ver el resultado de la exportación sin necesidad de tener el módulo document.
- Puesta transición de estados sin workflow.
- Pequeños ajustes de interfaz.
[IMP] l10n_es_aeat_mod303: Adaptado al nuevo mecanismo. Se elimina workflow anterior mediante un script de migración.

acysos committed:
[FIX] l10n_es_aeat: compatibilidad de la función export_to_boe para el modelo 340

Commit:
[FIX] l10n_es_aeat: Pequeños cambios para acomodar la funcionalidad del 347 y el 340.

Commit:
[IMP] l10n_es_aeat:

* Eliminación de la mayoría de warnings que da runbot.
* Usar librería unidecode en lugar de reemplazos manuales para las tildes y caracteres no soportados.

pedrobaeza committed:
[MIG] Movidos todos los módulos a __unported__
[MIG] Eliminados módulos no migrados a v7

pedrobaeza committed:
[FIX] l10n_es_aeat: New PEP8 checks
- Mover import de unidecode para evitar error

Commit:
[IMP] l10n_es_aeat:

* unificar campos en el módulo base + migrar a la nueva api.
* README.rst mejorado
* mejora de código
* usabilidad del atributo _aeat_number
* secuencia automatica
* control al instanciar los modelos
* correcciones necesarias para el correcto funcionamiento de los módulos

pedrobaeza committed:
[FIX] l10n_es_aeat_*: PEP8 + Refinamientos varios

pedrobaeza committed:
[IMP+FIX] l10n_es_aeat_*: Cálculo por códigos de impuestos generalizado + tratamiento residentes 216 y cálculo

pedrobaeza committed:
[MIG] l10n_es_aeat_mod303: Modelo 303 de la AEAT totalmente refactorizado

Presentación del Modelo AEAT 303
================================

* Incluye un motor genérico para obtener los totales de códigos de impuestos
  de manera parametrizada y con posibilidad de versionado por fechas
* Incluye un motor de parametrización de las exportaciones a formato BOE
  con posibilidad de versionado por fechas.
* Se incluyen las parametrizaciones de ambos motores para el 303.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Problema de permisos en l10n.es.aeat.tax.line

pedrobaeza committed:
[IMP] l10n_es_aeat_vat_prorrate: Enfoque definitivo

======================
AEAT - Prorrata de IVA
======================

Módulo para gestionar la prorrata del IVA en las declaraciones de la AEAT (por
el momento, en el modelo 303), según el artículo 92 de la Ley 37/1992, de 28 de
diciembre, del Impuesto sobre el Valor Añadido.

Uso
===

Realizando el modelo 303, se puede seleccionar si se desea aplicar prorrata de
IVA.

Prorrata general
----------------

En el caso de seleccionar "Prorrata general", aparece un nuevo campo llamado
"Porcentaje de prorrata de IVA" que puede ser rellenado a mano.

Además, cuando se realice el asiento de regularización, la parte proporcional
de la cuota no deducida se compensará como gasto.

Se ha incluido un calculador del porcentaje de prorrata, utilizable en 2 casos:

* Al comienzo de un nuevo ejercicio, para determinar el porcentaje provisional
  a aplicar hasta el último periodo de ese ejercicio.
* En la última declaración del ejercicio, para determinar el porcentaje
  definitivo de prorrata, y sobre el que se compensará la diferencia.

Problemas conocidos / Hoja de ruta
==================================

* La prorrata especial de IVA no está contemplada aún.

pedrobaeza committed:
[IMP] l10n_es_aeat/l10n_es_aeat_mod303: Crear asiento de regularización cuando se hace la declaración

* Generador del asiento de regularización con cargo a un proveedor "Agencia
  Estatal de Administración Tributaria" creado al efecto.

Configuración
=============

Para activar la creación del asiento de regularización en un modelo, hay que
poner en el modelo correspondiente el campo allow_posting a True, y establecer
en la configuración de impuestos los conceptos que se regularizarán con el
flag "to_regularize".

pedrobaeza committed:
[FIX] l10n_es_aeat_mod111: Eliminar periodo específico y corregir exportación

pedrobaeza committed:
[IMP] l10n_es_aeat_*: Cálculo de importes de impuestos por cadena en lugar de recordset + gestionar grupos de compañías

pedrobaeza committed:
[FIX] l10n_es_aeat: Remove tax code search limit

pedrobaeza committed:
[FIX] l10n_es_aeat: _formatNumber con 0 para positivos en lugar de espacio. Fixes #228

jcarlosmontoya committed:
[FIX]l10n_es_aeat_export_to_boe_button

PR related with: https://github.com/OCA/l10n-spain/issues/236

pedrobaeza committed:
[FIX] l10n_es_aeat: Default again to ' ' in positive numbers and allow other values in export config

Commit:
[8.0][FIX] l10n_es_aeat: Corrige creación de apuntes con importe en debe y haber a la vez.

pedrobaeza committed:
[FIX] l10n_es_aeat_*: Mejor herencia de impuestos

El anterior método utilizado, al tener una clase y un many2one genérico,
no permitía poner los datos en múltiples modelos hijos, ya que se repetían
los IDs, y cogía los datos de un modelo en otros.

Con este nuevo sistema, se sigue utilizando una tabla, pero con doble
índice: el modelo y el ID, parecido a un campo reference, pero gestionado
por código propio en el método de cálculo.

El 303 está adaptado a este nuevo sistema, y se incluye un script de
migración que conserve los datos.

pedrobaeza committed:
[IMP] l10n_es_aeat_report: Remove tax line when report is removed

antespi committed:
[FIX] Do not use hyphen in report sequence, only numbers
- Change field name: 'sequence' by 'name'
- Isolate report identifier number calculation in order to be inheritable
- Change methos name: _name_get by _report_identifier_get
- Remove sequence constraints

pedrobaeza committed:
[IMP] l10n_es_aeat: Don't split regularization lines for same account. Closes #305

If some tax lines have to been regularized, and they have the same account, previous
to this commit, they appear in separate lines. Now, all are merged in only one line
(if the balance sign is the same).

pedrobaeza committed:
[FIX] l10n_es_aeat: Sequence number multi-company

Roodin committed:
[FIX].l10n_es_aeat. Correccion de Warning en onchange de period_type

Roodin committed:
l10n_es_aeat,  l10n_es_aeat_mod349. Pequeñas mejoras

Roodin committed:
l10n_es_aeat. Pequeños retoques de código, según correcciones propuestas

pedrobaeza committed:
[FIX] l10n_es_aeat: Admit only valid character set on formatString

Commit:
[IMP] l10n_es_aeat_mod111: Nuevo formato de exportación del modelo 111 para los ejercicios 2016 y siguientes

pedrobaeza committed:
[FIX] l10n_es_aeat: Handle non string variables in _formatString - Fixes #375

cubells committed:
[IMP] Add tax amount to view

cubells committed:
[FIX] statusbar_colors not supported

pedrobaeza committed:
[MIG] l10n_es_aeat: Initial migration

antespi committed:
[MIG] l10n_es_aeat/303: Rest of the migration

pedrobaeza committed:
[FIX] l10n_es_aeat: Bump version

antespi committed:
[IMP] Tomar el último día del ultimo periodo para el asiento de regularización

antespi committed:
[8.0][IMP][l10n_es_aeat_mod347] Improve performance and other things (#317)

angelmoya committed:
[FIX] l10n_es_aeat: Make repeat_expresion work

pedrobaeza committed:
[IMP] l10n_es_aeat: Several warnings and imps

pedrobaeza committed:
[MIG] l10n_es_aeat/mod303: Migration to 10.0

pedrobaeza committed:
[FIX] l10n_es_aeat: Include direct statement account moves with taxes

When you fill a bank or cash statement with taxes for cancelling an expense,
those amounts weren't included on the reports, as the move type is
'liquidity', not 'receivable' or 'payable'. This fix includes also this type
on the search domain.

pedrobaeza committed:
[FIX] l10n_es_aeat: Correct constraint for multi-company

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Adapt new BOE 2017 format

pedrobaeza committed:
[IMP] l10n_es_aeat: Avoid rewrites on related fields

ORM launches a rewrite on origin table for related fields without
readonly flag. Preventing this marking it as readonly, we will
gain a bit in performance or avoid write errors.

pedrobaeza committed:
[FIX] l10n_es_aeat: Avoid multi-company error due to depends

Debido a que se utiliza una sola tabla para almacenar las líneas
de los impuestos de todos los modelos, hay una limitación en el ORM de Odoo
cuando se coloca el campo one2many de dichas líneas (tax_line_ids) como
dependencia en la definición del cálculo de un campo (entrada con
@api.depends), que recalcula los campos calculados de todos los modelos con el
mismo ID que el del registro en curso, lo que puede ser un problema en entornos
multi-compañía. Una solución a ello (aunque no evita el recálculo), es poner en
esos campos calculados `compute_sudo=True`.

pedrobaeza committed:
[IMP] l10n_es_aeat: Check previous declaration number on proper types

If the declaration is complementary or substitutive, there should be
a previous declaration number.

Roodin committed:
[FIX]. l10n_es_aeat. Fijado problema con referencai a _model al exportar

pedrobaeza committed:
[FIX] l10n_es_aeat_*: Conditional expression is reversed + 303 with conditional

gurneyalex committed:
[REF] i18n: Updating translation terms from weblate [ci skip]

pedrobaeza committed:
[FIX] l10n_es_aeat: Workaround for odoo/odoo#18438

Due to a bug in ORM that unlinks other tables' records, we have to avoid (0, 0, x) syntax

Reference: https://github.com/odoo/odoo/issues/18438

pedrobaeza committed:
[IMP] l10n_es_aeat: Add method for formatting BOE dates

misern2 committed:
[MIG] l10n_es_aeat_mod115: Migration  to 10.0

pedrobaeza committed:
[IMP] l10n_es_aeat: _register_hook without warning

Calling it with decorator @api.model_cr

oihane committed:
[MIG] l10n_es_aeat: Migration to version 11.0

Commit:
[FIX] l10n_es_aeat: ljust exige byte string de lenght 1

hugosantosred committed:
[FIX] l10n_es_aeat: Bytestring concatenation on file export to boe (Solves compatibility problems with python3) (#1)
* Adds tests for exporting files to boe with an export config (#2)

hugosantosred committed:
[FIX] l10n_es_aeat: Add onchange_chart_template_id to TestL10nEsAeatModBase._chart_of_accounts_create() in order to fix travis tests
* remove fax, field does not exist

JordiBForgeFlow committed:
[FIX] l10n_es_aeat: Fix BOE export (#718)

JordiBForgeFlow committed:
[MIG] l10n_es_aeat_mod349: Migration to 10.0

pedrobaeza committed:
[MIG] l10n_es_aeat_mod349: Migration to 11.0

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Improve tests for handling negative price units

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod303: Fix refinements

* No need to change test structure with create
* Optimize code
* Fix tests in l10n_es_vat_prorrate
* Translations
* Other minor things

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Date for regularization move on the end of period

JordiBForgeFlow committed:
migración de l10n_es_vat_book a 11.0

pedrobaeza committed:
[IMP] l10n_es_aeat: Add tax_exigibility option to tax mapping

With this, cash basis can be done in an easy way, and it's a base field
in account module.

Abranes committed:
[IMP] l10n_es_aeat: New export_type alphabetic

jvpascual committed:
[11.0] [IMP] l10n_es_aeat: Contact email added. (#866)

pedrobaeza committed:
[FIX] l10n_es_aeat: Adapt patch to P3

pedrobaeza committed:
[FIX] l10n_es_aeat_mod303: Handle negative amounts in purchase refunds

It includes tests for the new cases.

pedrobaeza committed:
[MIG] l10n_es_aeat: Migration to 11.0

- Standard procedure
- Adapt views
- Adapt tests
- README by fragments

pedrobaeza committed:
[MIG] l10n_es_aeat_mod303: Migration to 12.0

* Standard procedure
* Adapted tests
* README by fragments

pedrobaeza committed:
[FIX] l10n_es_aeat: Don't put company to AEAT partner

So it works for multi-company environments.

pedrobaeza committed:
[IMP] l10n_es_aeat_mod303: Add support for 2018 4T:

* BOE format
* Exonerated from 390

  * Fields
  * Mappings
  * BOE export

* README update
* Little change in AEAT engine
* Tests adapted + expanded

pedrobaeza committed:
[IMP] l10n_es_aeat: Add BOE file comparer

New feature for viewing a BOE exported file with each field properly mapped
to their export config line for showing the values.

pedrobaeza committed:
[IMP] l10n_es_aeat: Add active field to export configs

This way, we don't initially see all the subconfigs for each model and we can concentrate
on main export configurations. This is got through `active` field default behaviour.
Archived config are still directly searchable through the subconfig selector in the
configuration line.

All models export configs have been adapted to this new approach.

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat_mod347: Migration refinements + enhancements

* Add controller for automatic confirmation/rejection
* Improve data model
* Clean code
* Clean UI
* Spanish translation
* Tests
* ...

angelmoya committed:
[FIX] res_partner_bank.acc_type type of field

pedrobaeza committed:
[FIX] l10n_es_aeat: Create sequences when new company is created

When a new company is created, its sequences are not populated, so an Odoo service
restart is needed for being able to use AEAT models. With this, you don't need
this restart.

It includes a test with a hack for populating a model in the test itself for testing
complete flow.

pedrobaeza committed:
[FIX] l10n_es_aeat: Depends on l10n_es

This dependency is not strictly needed by the module, but the test base for other
modules needs the module, so let's put it explicitly

pedrobaeza committed:
[IMP] l10n_es_aeat: Move state mapping to base

pedrobaeza committed:
[FIX] l10n_es_aeat: Remove " from BOE files

It's not accepted by AEAT.

pedrobaeza committed:
[FIX] l10n_es_aeat_mod347: Proper variable assignation

So that inherited models are able to access this value.

pedrobaeza committed:
[IMP] l10n_es_aeat: Add chatter + activities to AEAT models

* Include chatter (mail.thread)
* Include activities (mail.activity.mixin)
* Track AEAT report state change
* Include new admin user in AEAT group by default

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (228 of 228 strings)

Translation: l10n-spain-12.0/l10n-spain-12.0-l10n_es_aeat
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-12-0/l10n-spain-12-0-l10n_es_aeat/es/

carlosdauden committed:
[IMP] l10n_es_aeat: Añadir campo aeat_anonymous_cash_customer y funcion parse_vat_info

carlosdauden committed:
[IMP] l10n_es_aeat: Añadir método que devuelve impuestos desde plantillas

sergio-teruel committed:
[FIX] l10n_es_aeat: fix report client action launch

acysos committed:
[ADD] l10n_es_aeat: certificates and soap. AEAT Partner check

[ADD] l10n_es_aeat: certificates and soap. AEAT Partner check

[ADD] l10n_es_aeat: certificates and soap. AEAT Partner check

[ADD] l10n_es_aeat: certificates and soap. AEAT Partner check

[ADD] l10n_es_aeat: certificates and soap. AEAT Partner check

MiquelRForgeFlow committed:
[UPD] Revert obsolete workaround

It was solved in v9 (see https://github.com/odoo/odoo/pull/19181/commits/9ae78328ec6fa0c1ba5c0a5f7583a565c7fa4e3f).

MiquelRForgeFlow committed:
[UPD] Remove deprecated from_string()

MiquelRForgeFlow committed:
[UPD] Remove deprecated decodestring and encodestring

MiquelRForgeFlow committed:
[IMP] Get taxes from templates instead of code description

MiquelRForgeFlow committed:
[IMP] l10n_es_aeat: black, isort

MiquelRForgeFlow committed:
[MIG] l10n_es_aeat: Migration to 13.0

pedrobaeza committed:
[IMP+FIX] l10n_es_aeat: Move get taxes method to company + cleaning in tests

This way, we can reuse it in other modules that don't inherit AEAT report like SII.

We also clean a non needed reading of the tax description field.

Tests were still using description method for populating data, so we switch to new
system and perform needed changes in dependent tests.

s

pedrobaeza committed:
[FIX] l10n_es_aeat: Avoid permission error posting invoice

Due to a weird chained operations, posting the invoice with billing user
permissions leads to an AccessError trying to read `l10n.es.aeat.mod303.report`.

And even more, this only happens on first test execution during install. Consecutive
test executions don't fail.

Possible reasons for this are:

- Registry state in SavepointCase in setUpClass
- New ORM API malfunction

Anyway, as doing this on UI or successive test executions don't fail, let's use
the shortcut of doing the post with sudo for avoiding the problem.

JordiBForgeFlow committed:
refactor html display of vat book

pedrobaeza committed:
[MIG+IMP] l10n_es_aeat_sii: Complete migration

- Redo the tax computation for efficiency and reliability
- Add test suite for checking obtained JSON for different tax configurations
- SII description computation reworked for new computed writable
- Added migration scripts

ozono committed:
[IMP] l10n_es_aeat: black, isort, prettier

ozono committed:
[MIG] l10n_es_aeat: Migration to 14.0

pedrobaeza committed:
[FIX] l10n_es_aeat: Incorrect translation

It was provoking an invalid extension in Windows.

pedrobaeza committed:
[IMP] l10n_es_aeat: Set some base data for other modules

- Being sure company is created with EUR
- Setting a resilient environment for USD

enriquemartin committed:
[IMP] l10n_es_aeat - unificar modelo aeat.sii.tax.agency

pedrobaeza committed:
[FIX] l10n_es_aeat: Allow to unlink regularization move when cancelling

From this version, there's no more "Allow cancelling entries" option
in journals to choose from, so we should force the removal of the entry
in these cases.

pedrobaeza committed:
[REF+IMP] l10n_es_aeat/l10n_es_aeat_sii_oca: Re-arrange certificate tests

- Put minimal certificate test in l10n_es_aeat
- Remove deprecation warning due to use of password as str.
- Extend properly tests in l10n_es_aeat_sii_oca for the SII part.

pedrobaeza committed:
[FIX] l10n_es_aeat: Don't force journal on refunds

Due to a newly introduce check, journal type must be compatible, so we
can't put in a misc journal any refund, but it's not needed to force it,
as reverse code handle this.

pedrobaeza committed:
[FIX] l10n_es_aeat: Don't screw VAT check

Since https://github.com/odoo/odoo/pull/68474, the tests are failing
as Odoo is not able to recognize the country for VAT check. We do the
writing for both values at the same time for avoiding the problem.

ValentinVinagre committed:
[FIX]l10n_es_aeat: the ._origin field is added to the search for accounts and taxes templates to make it compatible with objects of type NewID.

pedrobaeza committed:
[OU-ADD + IMP] l10n_es_aeat: Include record rule migration script + adjust rule

It includes the migration scripts for the module.

As the field is required, no need to check for empty company in the record rule.

etobella committed:
[IMP] l10n_es_aeat: Remove dependancy on PyOpenSSL

etobella committed:
[IMP] l10n_es_aeat: Fullfill data from the certificate

pedrobaeza committed:
[FIX] l10n_es_aeat: Non stored + archive ribbon

- Always assign value to non stored computed field
- Put archived ribbon ala v13 style
- Adopt it as maintainer

RodrigoBM committed:
[IMP] l10n_es_aeat_mod303_oss: Add field 124 and 127

pedrobaeza committed:
[MIG+IMP] l10n_es_aeat_mod303_oss: Migration to 13.0 + tests

- Standard procedure.
- Add tests for full coverage.
- Fix compatibility with `l10n_es_aeat_vat_prorrate` due to a problem
  with ormcache and virtual records.

ValentinVinagre committed:
[IMP] l10n_es_aeat: black, isort, prettier

ValentinVinagre committed:
[MIG] l10n_es_aeat: Migration to 15.0

ValentinVinagre committed:
[FIX] l10n_es_aeat: Fix development_status

ValentinVinagre committed:
[FIX] l10n_es_aeat: Added group group_erp_manager in the Account manager user for testing.

pedrobaeza committed:
Revert "[15.0][FIX] l10n_es_aeat: Added group group_erp_manager in the Account mana…"

ValentinVinagre committed:
[FIX] l10n_es_aeat: Using sudo for the model ir.model.data

etobella committed:
[IMP] l10n_es_aeat: allow to define the identification type

Co-authored-by: Ethan Hildick <ethan@studio73.es>

pedrobaeza committed:
[FIX] l10n_es_aeat: Proper position for anonymous check

We have a page for this info, and the previous place was breaking the
layout of such part.

pedrobaeza committed:
[IMP] l10n_es_aeat: Use dynamic parent for AEAT reports

As enterprise module `account_accountant` is using another root menu,
and other kind of hacks are not desirable, making a dynamic search over
the parent of another well know menu (Customers - menu_finance_receivables),
makes the trick for having the menu correctly placed in both editions.

Thanks to Martin Trigaux for suggesting this solution.

pedrobaeza committed:
[FIX] black formatting

LoisRForgeFlow committed:
[IMP] l10n_es_aeat: small adjustment in aeat tax agency view

sergio-teruel committed:
[FIX] l10n_es_aeat: Don't error on manually created export lines

pedrobaeza committed:
[FIX] l10n_es_aeat: Proper country code for Greece and old French colonies

When we have a Greek VAT number (ELXXXXXXXXX), `_parse_aeat_vat_info`
was incorrectly returning `GR, XXXXXXXXX, 02`. For AEAT, the country
code is always the original one [1], not the official ISO one.

We only need the ISO one for checking the inclusion in the European
union.

Thus, we make the mapping only for this inclusion check.

[1] https://sede.agenciatributaria.gob.es/static_files/Sede/Disenyo_registro/DR_300_399/archivos_20/DR_Anexo_349.pdf

pedrobaeza committed:
[FIX] l10n_es_aeat: Modify tests according country_code change

Followup of b4fe6bf8e28db3a19e5b08f2ef3e962cec7358df, as tests were
testing the exact incorrect behavior without context.

manuelregidor committed:
[FIX]l10n_es_aeat: Considerar modelo en comprobación de rango de fechas

manuelregidor committed:
[IMP]l10n_es_aeat: Base para mostrar numero errores en reports

pedrobaeza committed:
[IMP] l10n_es_aeat: Avoid deprecated class in tests

MiquelRForgeFlow committed:
[FIX] l10n_es_aeat_*: exceptions.Warning -> exceptions.UserError

pedrobaeza committed:
[FIX] l10n_es_aeat: Consider correctly French overseas

For French overseas territories
(https://en.wikipedia.org/wiki/Overseas_France), there are no
intracommunity fiscal rules, but the country code that AEAT requires
should be FR, as it can be seen here:

https://sede.agenciatributaria.gob.es/Sede/todas-gestiones/impuestos-tasas/declaraciones-informativas/modelo-347-decla_____racion-anual-operaciones-personas_/instrucciones/codigos-paises-territorios.html

Thus, we need to call the mapping function differently if you want to
convert `EL` to `GR` for getting the European countries than to pass
the proper country code.

This is done adding an extra parameter for saying if we want to include
such replacements, and only being executed when the check about European
country is rejected, on contrary than with Greece (EL), that should be
included as intracommunity.

The country code is now also obtained from the vat itself if the first
2 letter belongs to any country code.

TT38835

yajo committed:
[FIX] l10n_es_aeat: specify cryptography version

This module requires the specified cryptography version. Make that explicit in its manifest, now that thanks to https://github.com/odoo/odoo/pull/25549 we can specify the package version.

@moduon MT-1067

etobella committed:
[IMP] l10n_es_aeat: Fix cryptography dependancy

ramiadavid committed:
[MIG] l10n_es_aeat: Migration to 16.0

etobella committed:
[IMP] l10n_es_aeat + l10n_aeat_sii_oca: Add standard function for getting taxes

- New hook for managing changes on the function
- New field for tax_amount
- Call l10n_es_aeat method in SII instead of own (+ remove specific one)

etobella committed:
[IMP] l10n_es_aeat: Add standard function for getting taxes

New hook for managing changes on the function
New field for tax_amount

Commit:
[IMP] l10n_es_aeat: Hook for deprecated method

- Enables backwards compatibility and warns the user, to be removed in future migrations

Shide committed:
[FIX] l10n_es_aeat: Move `thirdparty_invoice` field to l10n_es_aeat

victoralmau committed:
[FIX] l10n_es_aeat: Move thirdparty_invoice + thirdparty_number field to l10n_es_aeat

eantones committed:
[IMP] l10n_es_aeat: Added support for tax groups in the calculation of tax info

xavier-bouquiaux committed:
[IMP] l10n_es_aeat: Improve install time

this commit : https://github.com/OCA/l10n-spain/commit/2dc5f549f91bda2f97f8b39624c1ea137ec18042~
introduces thirdparty_invoice as a compute store true on account_move table
on large database, the initial compute is too long
I created a hook and a migrate in order to fill it

ramiadavid committed:
[FIX] l10n_es_aeat: Remove 14.0 migration script

hugosantosred committed:
[FIX] l10n_es_aeat: add invisible company_id field in view

Without this field, the update of the 303 addon, throws an error
with the counterpart_account_id field that use the company_id in the
domain

pedrobaeza committed:
[FIX] l10n_es_aeat: Proper interval overlap checking in tax map

It includes unit tests.

victoralmau committed:
[IMP] l10n_es_aeat: SavePointCase > TransactionCase + Warning > UserError

warnings.warn("Warning is a deprecated alias to UserError.", DeprecationWarning)

pedrobaeza committed:
[FIX+IMP] l10n_es_aeat*: VAT prorate proper encapsulation + names + tests

pedrobaeza committed:
[FIX] l10n_es_aeat: No ir.model forbidden access

This model access is restricted to administrator, so we:

- sudo operations that require to access it.
- remove an auxiliary field computing it that is not used.

Fixes #3016

RodrigoBM committed:
[FIX] l10n_es_aeat: replace deprecated method.
- remove deprecated flush() method.
- replace deprecated invalidate_cache() method with env.invalidate_all().

Commit:
[FIX] l10n_es_aeat: fix broken views

pedrobaeza committed:
[FIX] l10n_es_aeat: Consider financial_type="other" as regular operations

Steps to reproduce the problem:

- PoS installed.
- Have a session with only payment methods with outstanding accounts (no cash payments received).
- Close and post session.
- The resulting journal entry is of type = "Other".

Such entries, even if having taxes, won't be included in the AEAT 303 model,
as the type = "Other" is not considered till now. That's because it can't
be deduced if the operation is regular or not.

Let's assume that such operations are always regular (as 99.9% of the times will be),
for not having differences in the amounts of the report vs balances in the tax.

TT44503

etobella committed:
[IMP] l10n_es_aeat: Make it work with multicompanies

Commit:
[16.0][IMP] l10n_es_aeat: Add tax agency

aritzolea committed:
[IMP] l10n_es_aeat: Add country ISO code mapper function

Fulfills the need (seen in l10n_es_aeat_mod369) to obtain the ISO code of a country based on the country code stored in Odoo. It is an inverse function of _map_aeat_country_code.

czea committed:
l10n_es_aeat: No ir.model forbidden access, sudo in domain calc

Fixes one more remaining unauthorized access to ir.model
On export_config_id field, domain calculation searched on ir.model
Added sudo() to avoid unauthorized access.

fixes #3016

pedrobaeza committed:
[FIX] l10n_es_aeat: Convert COM countries to France

COM (collectivité d'outre-mer -
https://es.wikipedia.org/wiki/Colectividad_de_ultramar)
are also considered France for the AEAT, so we need to map the existing
countries to FR.

TT46789

EmilioPascual committed:
[IMP] l10n_es_aeat_report: Order aeat reports by date start

@moduon MT-4519

ramiadavid committed:
[IMP] l10n_es_aeat: pre-commit auto fixes

ramiadavid committed:
[MIG] l10n_es_aeat: Migration to 17.0

ramiadavid commented:
@pedrobaeza como te comenté, esta es mi propuesta para los cambios en odoo 17, he creado dos modelos una para las plantillas de los impuestos y otro para las cuentas, yo personalmente creo que es mejor así que escribir los xml_id directamente en los mapeos, por lo menos visualmente es mejor si queremos hacer algún cambio.

Los registros de esos dos modelos se crean con los CSV del plan contable, estos registros se crean/actualizan cuando se instala/actualiza el módulo de modo que si se crean nuevos impuestos en el plan contable de la localización española, al actualizar el módulo ya aparecerán en el listado para poder mapearlos.

ramiadavid commented:
@pedrobaeza como te comenté, esta es mi propuesta para los cambios en odoo 17, he creado dos modelos una para las plantillas de los impuestos y otro para las cuentas, yo personalmente creo que es mejor así que escribir los xml_id directamente en los mapeos, por lo menos visualmente es mejor si queremos hacer algún cambio.

Los registros de esos dos modelos se crean con los CSV del plan contable, estos registros se crean/actualizan cuando se instala/actualiza el módulo de modo que si se crean nuevos impuestos en el plan contable de la localización española, al actualizar el módulo ya aparecerán en el listado para poder mapearlos.

HaraldPanten commented:
/ocabot migration l10n_es_aeat

HaraldPanten commented:
@ramiadavid Si haces rebase, ya te debería coger bien la dependencia 👍

ValentinVinagre commented:
@ramiadavid no veo mal la solución para reducir el impacto sobre todo el sistema de contabilidad española. Pero ¿Se ha tenido en cuenta también estos cambios como aplicarían al account.chat.update de oca? ¿Sería necesario realizar un módulo de extensión para que tenga en cuenta este cambio?

ramiadavid commented:
No es un módulo que haya tocado y no conozco mucho cómo funciona, pero seguro que habrá que hacer cambios

ramiadavid commented on l10n_es_aeat/tests/test_l10n_es_aeat_export_config.py:
```diff
@@ -163,7 +163,7 @@ def test_export_config_file(self):
             }
         )
-        new_report = self.env["l10n.es.aeat.report"].new({"name": "Test Report"})
+        new_report = self.env["l10n.es.aeat.report"]
```
Después de hacer el rebase ha comenzado a darme este error:
`TestL10nEsAeatExportConfig.test_export_config_file
2024-02-16T19:04:19.9567010Z Traceback (most recent call last):
2024-02-16T19:04:19.9567904Z   File "/__w/l10n-spain/l10n-spain/l10n_es_aeat/tests/test_l10n_es_aeat_export_config.py", line 166, in test_export_config_file
2024-02-16T19:04:19.9568806Z     new_report = self.env["l10n.es.aeat.report"].new({"name": "Test Report"})
2024-02-16T19:04:19.9569394Z   File "/opt/odoo/odoo/models.py", line 6396, in new
2024-02-16T19:04:19.9569844Z     record._update_cache(values, validate=False)
2024-02-16T19:04:19.9570318Z   File "/opt/odoo/odoo/models.py", line 5996, in _update_cache
2024-02-16T19:04:19.9570810Z     cache.set(self, field, value, check_dirty=False)
2024-02-16T19:04:19.9571241Z   File "/opt/odoo/odoo/api.py", line 999, in set
2024-02-16T19:04:19.9571605Z     record_id = record.id
2024-02-16T19:04:19.9572084Z AttributeError: 'l10n.es.aeat.report' object has no attribute 'id'`

Por lo visto en la última versión de odoo17 no se puede hacer .new() a un AbrstactModel ya que da error, quitando esta parte el test finaliza correctamente, así que entiendo que estará correcto.

HaraldPanten commented:
Hola David,

Primero de todo, agradecerte la iniciativa de migración. Nos ha tocado asumir un cambio global en el sistema por lo de la desaparición de las `template` y toca repensar el sistema a nivel de localización española.

Me toca hacer de poli malo ahora... Quizá se me escape algo, pero has tenido en cuenta los comentarios de Pedro en este comentario? --> https://github.com/OCA/l10n-spain/issues/3382#issuecomment-1895323206

Salvo error por mi parte, no acabo de ver claro que estemos todos en sintonía.

@pedrobaeza ¿Es el planteamiento de David al que te referías con ello? No lo tengo claro... Si queréis podemos organizar meet por el Discord de AEOdoo para aclarar el planteamiento, y si hace falta heredamos nosotros (Sygel) la migración?

Saludos.

ramiadavid commented:
@HaraldPanten el planteamiento mio es similar al de @pedrobaeza, es solo una propuesta y hay que analizar pros y contras, 

El planteaba en los mapeos cambiarlo por campos tipo texto y escribir el xml_id del template, pero pienso que eso es muy poco visual si algun usuario tiene que tocar alguno, lo que yo planteo es volver a crear esos templates en un modelo nuevo con dos simples campos Nombre y xml_id de esta forma es mas sencillo de seleccionar desde una lista

Esos templates se crean extrayendo los datos de los .csv de forma que si se añaden mas impuestos al actualizar el módulo ya aparecen en la lista

Podéis continuar con ello vosotros si quereis, por mi no hay problema

ValentinVinagre commented:
> @HaraldPanten el planteamiento mio es similar al de @pedrobaeza, es solo una propuesta y hay que analizar pros y contras,
> 
> El planteaba en los mapeos cambiarlo por campos tipo texto y escribir el xml_id del template, pero pienso que eso es muy poco visual si algun usuario tiene que tocar alguno, lo que yo planteo es volver a crear esos templates en un modelo nuevo con dos simples campos Nombre y xml_id de esta forma es mas sencillo de seleccionar desde una lista
> 
> Esos templates se crean extrayendo los datos de los .csv de forma que si se añaden mas impuestos al actualizar el módulo ya aparecen en la lista
> 
> Podéis continuar con ello vosotros si quereis, por mi no hay problema

Hola @ramiadavid ,
Primero de todo merci por el trabajo 😄 . Dejo unos comentarios según he visto lo que comenta pedro y esta propuesta.

En efecto esta manera sería más sencillo a nivel de persona técnica, pero... ¿y para una persona funcional y mantenimiento? Pienso que la creación de 2 modelos para el mantenimiento puede ser más problemático ¿Por que motivos?

* 2 modelos nuevos que mantener (indexados para una mejor búsqueda). Por otro lado, se añaden búsquedas extras debido a que se tendrán que buscar previamente los records asociados y después el objeto en cuestión.
* ¿Como se controlan los errores en esos 2 modelos nuevos?
* A nivel de migración sería mantener modelos nuevos con su propio sistema adaptado a odoo para mantenimiento los datos, l veo más mantenimiento en el futuro. En cambio la manera de texto no tendría casi impacto ninguno.
* Se repite el dato en el sistema, si ya existe el dato en el impuesto dado por el xml_id ¿para que repetirlo?
* Mantenimiento de las 2 tablas según los giros que de Odoo. El sistema de xml_id lo veo más sencillo y que difícilmente cambiará.
* A nivel funcional, será más sencillo de mapear/mantener. Si que es cierto por eso que los xml-id's tendrán que tener buenos nombres, si no será aún más complejo de entender.
* Si odoo actualiza impuestos y/o cuentas (mediante su update) habrá que realizar código extra para que esos cambios pasen a las tablas de mapeos. Por no contar el account_chart_update ¿Que pasa si odoo crea un impuesto nuevo que después no quieres utilizar porque te lo ha creado mal y tienes que volver a mapear el antiguo? Veo más sencillo solo cambiando de external_id.
* Si se quieren crear impuestos nuevos y mapear por código en general este método en mi opinión lo veo más sencillo.
* A nivel de impacto, únicamente afectaría en la manera de búsqueda en los listados


Como bien se indica esto es una propuesta y vale la pena revisar ambos puntos de vista. 
¿Como lo veis según los puntos indicados?

etobella commented:
@ramiadavid Desde mi punto de vista, se ha intentado transformar lo existente para reaprovechar el código antiguo. En este caso creo que es mejor hacer un refactoring. Si vamos añadiendo capas para que se adapte a lo original lo único que haremos es crear un monstruo. Realizarlo tal y como comentan @HaraldPanten @ValentinVinagre  y @pedrobaeza es adapatarse a la lógica nueva de odoo y, a la larga, minimizar el código y las posibles fuentes de error.

My 2 cents

HaraldPanten commented:
> @ramiadavid Desde mi punto de vista, se ha intentado transformar lo existente para reaprovechar el código antiguo. En este caso creo que es mejor hacer un refactoring. Si vamos añadiendo capas para que se adapte a lo original lo único que haremos es crear un monstruo. Realizarlo tal y como comentan @HaraldPanten @ValentinVinagre y @pedrobaeza es adapatarse a la lógica nueva de odoo y, a la larga, minimizar el código y las posibles fuentes de error.
> 
> My 2 cents

Coincido con vosotros. Sale más a cuenta adaptarse a los cambios del software (a veces nos gustarán más y otras veces menos) que recrear lógicas antiguas; ya que eso nos llevará a tener que ampliar mucho más el código en cada una de las versiones y en cada uno de los módulos relacionados.

pedrobaeza committed:
[MIG] l10n_es_aeat: Finish migration

- Switch to XML-ID names models
- Adapt API and calls
- Some other minor adjustments
- Translations

pedrobaeza commented:
Hola, acabo de empujar un commit (aparte de hacer un rebase) que incluye la visión que os quería transmitir, teniendo un modelo simple con solo el XML-ID como `name`, y que cada submódulo añada las entradas que correspondan. Así, no ponemos trabas a si está pre-creado o no. Incluso puede favorecer un poco más la manipulación manual, ya que habrá que crear el impuesto, ponerle un XML-ID con formato compatible, y ya añadir el registro en el mapeo, y eso no se perderá con las actualizaciones.

Enseguida empujo el 303, que lo he hecho también para probar que funciona. Ahora reviso el error de los tests, que en local no salía.

pedrobaeza commented:
Y aquí está el 303: https://github.com/OCA/l10n-spain/pull/3486

HaraldPanten commented:
Gracias Pedro, a partir de mañana empezamos a revisar y a darle caña a la localización 💪🏻💪🏻

HaraldPanten reviewed (approved):
Probado en entorno local con el PR del modelo 303 y no he detectado ningún comportamiento inesperado. A falta de probarlo en producción con casos reales.

LGTM

pedrobaeza commented:
@ramiadavid te parece correcto el diseño?

ramiadavid commented:
Estoy fuera unos dias y no puedo probarlo, pero si a vosotros os parece bien, por mi está bien, me interesa tirar esto para adelante lo antes posible

pedrobaeza commented:
Venga, pues gracias por la confianza. Vamos con ello para desbloquear modelos, que ya no van a tener mucho trabajo.

Creé un script para transformar fácilmente los mapeados de v16 en los de v17:

https://gist.github.com/pedrobaeza/f68bf5e15cd18f5c1d1d5813df7be935

/ocabot merge nobump

etobella reviewed (approved):
Gracias @pedrobaeza por desbloquearlo :heart:
